src/main/java/vivatech/init/VivatechBlocks.java

```diff
@@ -21,9 +32,13 @@
         ENERGY_CONDUIT = new EnergyConduitBlock();
         COAL_GENERATOR = new CoalGeneratorBlock();
         CRUSHER = new CrusherBlock();
-        ELECTRIC_FURNACE = new ElectricFurnaceBlock();
         ENERGY_BANK = new EnergyBankBlock();
         PRESS = new PressBlock();
+        
+        ELECTRIC_FURNACE = new LinkedHashMap<MachineTiers, ElectricFurnaceBlock>();
```

Comment:
What do we gain from a map here? What do we gain from the definite ordering from LinkedHashMap? (consider: we burn memory by overlaying a linked list onto a hashmap, and we burn memory on an extra array for the hashmap instead of just an array or ArrayList). Especially for n=3, an O(n) search to find the correctly-tiered machine isn't really a problem worth optimizing.